container/lxd/initialisation_linux.go

```diff
@@ -156,24 +157,107 @@ var df = func(path string) (uint64, error) {
 }
 
 var configureLXDBridge = func() error {
-	client, err := ConnectLocal()
+	server, err := ConnectLocal()
 	if err != nil {
 		return errors.Trace(err)
 	}
 
-	status, err := client.ServerStatus()
 	if err != nil {
 		return errors.Trace(err)
 	}
 
 	// If LXD itself supports managing networks (added in LXD 2.3) we can allow
 	// it to do all of the network configuration.
-	if shared.StringInSlice("network", status.APIExtensions) {
-		return lxdclient.CreateDefaultBridgeInDefaultProfile(client)
+	if server.HasExtension("network") {
+		return createDefaultBridgeInDefaultProfile(server)
 	}
 	return configureLXDBridgeForOlderLXD()
 }
 
+type networkClient interface {
+	GetNetwork(name string) (network *api.Network, ETag string, err error)
+	CreateNetwork(network api.NetworksPost) (err error)
+	UpdateProfile(name string, profile api.ProfilePut, ETag string) (err error)
+	GetProfile(name string) (profile *api.Profile, ETag string, err error)
+}
+
+func checkBridgeConfig(client networkClient, bridge string) error {
+	n, _, err := client.GetNetwork(bridge)
+	if err != nil {
+		return errors.Annotatef(err, "LXD %s network config", bridge)
+	}
+	ipv6AddressConfig := n.Config["ipv6.address"]
+	if n.Managed && ipv6AddressConfig != "none" && ipv6AddressConfig != "" {
+		return errors.Errorf(`juju doesn't support ipv6. Please disable LXD's IPV6:
+
+	$ lxc network set %s ipv6.address none
+
+and rebootstrap`, bridge)
+	}
+
+	return nil
+}
+
+// CreateDefaultBridgeInDefaultProfile creates a default bridge if it doesn't
+// exist and (if necessary) inserts it into the default profile.
+func createDefaultBridgeInDefaultProfile(client networkClient) error {
+	/* create the default bridge if it doesn't exist */
+	n, _, err := client.GetNetwork(network.DefaultLXDBridge)
+	if err != nil {
```

Comment:
does LXD give us a clear NotFound error, vs a "could not connect to LXD" sort of error?